database/temp_transformation.go

```diff
@@ -0,0 +1,48 @@
+package database
+
+import (
+	"encoding/json"
+	"fmt"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/kit/score"
+)
+
+// This file is mean to temporarily transform BuildInfo and ScoreObjects strings to new score.Results type
+// so that we can continue to use the old database for testing.
+// TODO(meling) Remove this file and its uses when migration to new format is done.
+
+func transform(submissions ...*pb.Submission) {
+	for _, sub := range submissions {
+		buildInfo := unmarshalBuildInfo(sub)
+		scores := unmarshalScores(sub)
+		if buildInfo != nil && scores != nil {
+			sub.BuildInfo = buildInfo
+			sub.Scores = scores
+			sub.OldBuildInfo = ""
+			sub.ScoreObjects = ""
+		}
+	}
+}
+
+func unmarshalBuildInfo(submission *pb.Submission) *score.BuildInfo {
+	if submission.OldBuildInfo == "" {
```

Comment:
But won't this `OldBuildInfo` always be empty in the old database? I mean, a Submission in the old database will still have its jsonified build info in the `buildInfo` column, and every old Submission will have an empty `oldBuildInfo` . Are we supposed to rename the column manually in the old database or am I missing something completely?

Reply:
Another idea is to just load the database and use these transformations and write back things to a new database file that uses the new fields/columns. (As a one time task…)

Reply:
Yea, you sure can edit a column name in the database file, probably just worth mentioning somewhere that this is a necessary step to get things compatible.